### ISSUE TITLE
Update fdroid release

### ISSUE THREAD

**Issue by leg7** [grote/Transportr#826]
The last fdroid release was 4 years ago and even using the custom repository in the readme doesn't fix this.

Can the fdroid version be updated?

**Comment by julianfairfax** [comment]
This is the supposed solution: https://gitlab.com/fdroid/fdroidserver/-/issues/1004#note_1036153553

**Comment by leg7** [comment]
So from what I got a solution is being worked on to replace the google maps dependency with maplibre?

Is google maps a proprietary blob included in the apk or is transportr just using the api? Would it be feasible to get rid of the proprietary dependency upstream?

**Comment by julianfairfax** [comment]
> So from what I got a solution is being worked on to replace the google maps dependency with maplibre?
> 

I don't know if it's being worked on but it has been suggested.
> Is google maps a proprietary blob included in the apk or is transportr just using the api? Would it be feasible to get rid of the proprietary dependency upstream?

Mapbox is dragging it in. I suppose it would be possible to fix that upstream, but if switching to MapLibre works, that's probably a better solution.

**Comment by ialokim** [comment]
Yes, we're aiming to switching to MapLibre since new versions of Mapbox itself ar not compatible with free software anymore. Still default MapLibre currently has the same problem of depending on proprietary parts at compile time, which is not accepted by F-Droid. See the upstream issue at https://github.com/maplibre/maplibre-gl-native/issues/379 and #668 for our part, switching to MapLibre.

**Comment by jakoblm** [comment]
The issue has now been pushed again in the MapLibre repo: https://github.com/maplibre/maplibre-gl-native/issues/379

**Comment by jakoblm** [comment]
How about Transportr switching to the alternative AnyMaps (https://github.com/johan12345/AnyMaps) or osmdroid (https://github.com/osmdroid/osmdroid)? Obviously both use only free software and are accepted by F-Droid without any problems. See the following comment: https://github.com/maplibre/maplibre-gl-native/issues/379#issuecomment-1250773381

**Comment by 1Maxnet1** [comment]
A short time workaround would be to offer stable releases over a third party repository (e.g. https://android.izzysoft.de/repo/info offers to include apps in his repository). This would enable users to add the repository and use a more modern app version.

**Comment by ialokim** [comment]
> How about Transportr switching to the alternative AnyMaps (https://github.com/johan12345/AnyMaps) or osmdroid (https://github.com/osmdroid/osmdroid)? Obviously both use only free software and are accepted by F-Droid without any problems. See the following comment: [maplibre/maplibre-gl-native#379 (comment)](https://github.com/maplibre/maplibre-gl-native/issues/379#issuecomment-1250773381)

@jakoblm AnyMaps is just a wrapper over MapLibre with different API and osmdroid is a completely different kind of thing based on raster tile maps instead of vector maps. It is not realistic to change the API or even the complete map library just because of this issue which should be fixable upstream at MapLibre.

**Comment by ialokim** [comment]
> A short time workaround would be to offer stable releases over a third party repository (e.g. https://android.izzysoft.de/repo/info offers to include apps in his repository). This would enable users to add the repository and use a more modern app version.

This is actually a good idea, maybe @Izzysoft could tell us what he'd need from us? (I'd prefer his repo over another private one since it is probably more likely that people have it added to their F-Droid client already.)

Perhaps we should also disable Transportr on the F-Droid repo completely until we fixed the issue to avoid having people installing outdated versions like #831.

**Comment by IzzySoft** [comment]
Everything needed is in place, so I've just added Transportr to my repo as requested (effective with the next sync around 6 pm UTC).

**Comment by starsforeveryone** [comment]
I strongly advise that F-Droid is continued to be supported, and the newest version can be found there without any third party repositories.

The whole point of Transportr was to create a FOSS (free and open source software) app. F-Droid symbolizes this very idea of supporting freedom and privacy. I believe the author of Transportr shares these values. We already have a world-wide map created by nerds from around the world in their free time, Open Street Map, which would be an excellent replacement. What would be the point of these nerds doing this job for the past 18 years when we then don't use it and stick with Google instead? We're better than that.

Therefore, we should get rid of Google Maps and replace it with Open Street Map through osmdroid. :)

**Comment by IzzySoft** [comment]
@starsforeveryone that's known and being worked on. The culprit is a dependency of a dependency ([see here](https://gitlab.com/fdroid/fdroidserver/-/issues/1004#note_1030833616): it's not Google Maps but Mapbox being used as a wrapper for the map backend, but Mapbox unfortunately depends on GMS for location), and a replacement needs to be implemented (e.g. maplibre). As pointed out by @1Maxnet1 the listing in my repo is intended as a temporary work-around until the issue is solved (though I'd have no problems with keeping Transportr in my repo if that is wanted).

**Comment by starsforeveryone** [comment]
Thanks for clearing that up! :)

**Comment by David-Else** [comment]
I just heard about this software, went to install it from Fdroid, and to my horror found it looked like abandon-ware! I nearly gave up, but spent the extra time to research and came to this thread. I am sure 90% of people would not have bothered.

Will a proper version be coming to Fdroid this year, or should I look for other software? Cheers.

**Comment by jakoblm** [comment]
At the moment, that depends on whether a solution is found in Maplibre that meets the specifications at F-Droid. 
The Maplibre team has now asked for help to find a good solution soon. Wouldn't some experts from Transportr be willing to cooperate with Maplibre? See details here: https://github.com/maplibre/maplibre-gl-native/issues/379#issuecomment-1304451578

**Comment by genofire** [comment]
Now landed in version 10.0.0

**Comment by mnalis** [comment]
So, now that new Maplibre (which no longer depends on non-free GMS) has been released, we need to wait for new version of Transportr  to upgrade to it, and then it might be included in f-droid again?

**Comment by Altonss** [comment]
> So, now that new Maplibre (which no longer depends on non-free GMS) has been released, we need to wait for new version of Transportr to upgrade to it, and then it might be included in f-droid again?

Yes a new version of Transportr needs to be published (integrating Maplibre instead of Mapbox) for this newest version to appear in F-Droid (Transportr is still included in F-Droid, but sadly only a very old version).

**Comment by tuxayo** [comment]
> a new version of Transportr needs to be published

And developed ^^"

By any chance, has anyone taken a first look at how much work could it be for the switch? Hopefully the Mapbox version of Transportr isn't too old, otherwise there would also be additional upgrade work.

**Comment by Altonss** [comment]
> > a new version of Transportr needs to be published
> 
> And developed ^^"
> 
> By any chance, has anyone taken a first look at how much work could it be for the switch? Hopefully the Mapbox version of Transportr isn't too old, otherwise there would also be additional upgrade work.

A lot of work has already been done by @ialokim on this branch: https://github.com/grote/Transportr/tree/maplibre

**Comment by genofire** [comment]
No pressure, but is there any update?

**Comment by Altonss** [comment]
> > No pressure, but is there any update?
> 
> I tried building the latest maplibre branch revision ([29fbd55](https://github.com/grote/Transportr/commit/29fbd553f8859d30114fa1dd99922635755e170f)) from source with termux but the build fails. Someone needs finish the migration. I can give it a shot next week after my finals if it's not too complicated as I know nothing about this codebase or maplibre.

When I last tried, I was able to build the maplibre branch of @ialokim without an issue on Android Studio

**Comment by Jowat97** [comment]
As far as I can tell, there is already a pull request to fix this issue: https://github.com/grote/Transportr/pull/880
Is there anything from preventing or is any further help needed? Please feel free to ask. It would be amazing to have this app back on F-Droid.

**Comment by ialokim** [comment]
Even the old versions of Transportr are now disabled: https://gitlab.com/fdroid/fdroiddata/-/commit/0b38fbe548cfa419ec3c94c1286baaf9eec031b6

Indeed, the PR to bring us back to F-Droid is already in the pipeline: #880

**Comment by omnivagant** [comment]
Apparently I've been using an old version without really realizing, happy that f-droid doesn't remove disabled apps. I noticed because I wanted to suggest the app to some friends and went into the f-droid app to fetch an f-droid link and just couldn't find it.

This map function seems very neat, although I've never felt that I've missed it, I will probably use it when an updated version is available in the f-droid repo again.

However, is it possible to disable this map function and/or would it work with offline maps? I did see #126 and understand why offline traffic planning would be hard, but at least map data shouldn't have to be fetched if maps are already downloaded.

This also brings me to an even less related issue, that the varoius offline map apps don't share map data so you often need duplicates if you switch beetween map apps, say OsmAnd~ and Organic Maps.

Anyways, thank you for a very useful app even when several years out of date.

**Comment by licaon-kter** [comment]
https://gitlab.com/fdroid/fdroiddata/-/commit/c47b332f453092521ef1d38ae0a9bb065da2d0f0


/close this :)

**Comment by rugk** [comment]
Its on F-Droid is not it?

**Comment by licaon-kter** [comment]
@rugk https://gitlab.com/fdroid/wiki/-/wikis/FAQ#how-long-does-it-take-for-my-app-to-show-up-on-website-and-client

**Comment by rugk** [comment]
No I mean, I already saw it. Updated 4 days ago.

**Comment by ialokim** [comment]
> No I mean, I already saw it. Updated 4 days ago.

Yes, but this is from IzzyOnDroid's repo that has less strict inclusion criteria: https://apt.izzysoft.de/fdroid/index/apk/de.grobox.liberario We've added Transportr there when it got excluded from F-Droid's main repo.

Good news though: Transportr has already been built successfully during the current F-Droid build run: https://monitor.f-droid.org/builds/log/de.grobox.liberario/124#site-footer

**Comment by IzzySoft** [comment]
@julianfairfax @ialokim @Altonss as "since back then" Transportr is also available at IzzyOnDroid, maybe you want to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) and link there, next to the other two? At IzzyOnDroid it's even confirmed as RB :wink:
